### PR TITLE
Improve error messages

### PR DESCRIPTION
**Pull request by sommd**
Mostly just returning `Post Not Found` instead of `Malformed request` if the post doesn't exist.

Might improve some other error messages if I have time, or are they intentionally vague?

### CODE REVIEW

File: backend/namespaces/dummy.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def post(self):
         }
 
     @dummy.response(200, 'Success')
-    @dummy.response(400, 'Malformed Request')
+    @dummy.response(400, 'Malformed Request / Post Not Found')
```
Comment (DarkPurple141) [inline backend/namespaces/dummy.py]: If the request is valid, but the post cannot be found I'd suggest, 404 is a better status code.

**Review by DarkPurple141 — CHANGES_REQUESTED** [review]
Leave 400 as bad request, if you want to add the below functionality, add it under 404.

**Review by zainafzal08 — APPROVED** [review]
Looks good! Just address Alex's point before doing a cheeky PR

File: backend/namespaces/dummy.py (hunk 1 above)
Comment (sommd) [reply backend/namespaces/dummy.py]: I 100% agree. This is what I had at first, but flask_restplus decides to be helpful and suggests that the endpoint is wrong, which is why I decided to go with 400. I can change it to 404, but then you get this:

```json
{
    "message": "Post Not Found. You have requested this URI [/dummy/post] but did you mean /dummy/post or /dummy/post/like or /dummy/post/unlike ?"
}
```

Comment (DarkPurple141) [reply backend/namespaces/dummy.py]: So it does include the message at first though. Maybe extend the message to be:

`Post Not Found: This was a valid request, but the database could not find the post you're looking for. Else, you may also have hit the wrong endpoint.`

Verbose, but will clarify the above a bit better.

I think the restplus would work if the route was post/<id> but it ain't. Oh wellz.

**Review by DarkPurple141 — APPROVED** [review]

File: backend/namespaces/dummy.py (hunk 1 above)
Comment (sommd) [reply backend/namespaces/dummy.py]: Found a solution.

```python
app.config['ERROR_404_HELP'] = False
```

It gets rid of the message, but it does it for everything. Shouldn't be too much of a problem, right?

**Comment by DarkPurple141** [comment]
That works too.

**Comment by sommd** [comment]
Going to keep improving error messages when I have free time. Should reduce some headache for students.

**Pull request merged**